### ISSUE TITLE
v1.34.0 breaks screen reader accessibility in firewall rules tables

### ISSUE THREAD

**Issue by tech10** [linode/manager#7483]
Version 1.34.0 prevents a screen reader from properly reading and navigating the firewall rules tables. My specific use case is on NVDA using Google Chrome, though I believe the screen reader and web browser being used would be irrelevant to this particular issue.

To reproduce this, create a firewall and add some rules, either inbound or outbound should work for this. Using a screen reader, the only buttons accessible when arrowing through the tables will be labeled as View Details. Tabbing through the elements does seem to provide a bit more information and a few more buttons, which can't be activated with the screen reader, and the rule being viewed isn't visible to the screen reader.

With the firewall rules interface as it currently is, this makes things far more limited in their use, so far as accessible navigation with a screen reader goes.

**Comment by tech10** [comment]
@tiffwong Since firewalls are now out of beta, I thought I would see if there are any updates to fix this issue soon. Using the Linode CLI to edit firewall rules isn't always desired over the cloud manager, and as it is, it makes firewalls look unattractive to screen reader users with the lack of accessibility.

**Comment by tiffwong** [comment]
@tech10 Thanks so much for bringing this up again, we're prioritizing this and hope to have a solution soon!

**Comment by tech10** [comment]
#8735 does a good job of resolving this. It's definitely useable now, though the layout does look a bit different than the rest of the cloud manager. Also, when moving through the list with arrows using NVDA, in browse mode, the screen reader does tend to state the instructions for dragging continually with every option I move through.

Though it is useable as it is, it isn't perhaps as intuitive as it could be, since you have to enter focus mode before you're able to do anything, with NVDA, at least. This may not be quite as useable for some people as it is for me.

If this is to be the standard layout for the Firewall section, this issue can be closed. If not, and there's more work to be done to improve accessibility with screen readers, I would recommend keeping this issue open, for the time being. I would welcome any questions regarding possible accessibility improvements here.

**Comment by mjac0bs** [comment]
@tech10 Thank you for letting us know that recent changes resolved the issues you were encountering. We also appreciate your feedback on the areas where there is still room for improvement. Some of the unintuitive behavior you have mentioned is built into the library we use for drag and drop functionality; while we don't have further improvements at this time, it's possible an update to that library could help us improve accessibility in the future.

I'm going to go ahead and keep this issue open until we follow up on another firewall accessibility issue we found when testing accessibility for #8735. Options in select drop downs in the Add Inbound/Outbound Rule form and the Edit Rule form are not correctly read out ("Blank") on Windows with an NVDA screen reader. We have an internal ticket created to address this so users know the firewall form field options available to them.
